### PR TITLE
explain stream states

### PR DESCRIPTION
**Pull request by kazuho**

### CODE REVIEW

File: draft-ietf-quic-reliable-stream-reset.md [hunk 1]
```diff
@@ -158,6 +158,28 @@ a RESET_STREAM_AT frame. Similary, the Reliable Size of the RESET_STREAM_AT
 frame does not prevent a QUIC stack from delivering data beyond the
 specified offset to the receiving application.
 
+## Stream States {#stream-states}
+
+In terms of stream state transitions ({{Section 3 of RFC9000}}, the effect of
+RESET_STREAM_AT frame is equivalent to that of the FIN bit. This is because both
+the RESET_STREAM_AT frame and the FIN bit serve the same role: signalling the
```
Comment (marten-seemann) [inline draft-ietf-quic-reliable-stream-reset.md:165]: I believe this is the more common spelling in American English:
```suggestion
the RESET_STREAM_AT frame and the FIN bit serve the same role: signaling the
```

File: draft-ietf-quic-reliable-stream-reset.md [hunk 2]
```diff
@@ -158,6 +158,28 @@ a RESET_STREAM_AT frame. Similary, the Reliable Size of the RESET_STREAM_AT
...
+amount of data to be delivered.
+
+On the sending side, when a RESET_STREAM_AT frame is sent, the sending part of
+the stream enters the "Data Sent" state. Once the RESET_STREAM_AT frame and all
+stream data up to the Reliable Size are acknowledged, the sending part of the
+stream enters the "Data Recvd" state. Transition from "Data Sent" to "Data
+Recvd" happens immediately when the application resets a stream and if all bytes
+up to the specified Reliable Size have been sent and acknowledged already.
+Conversely, the transition might take multiple roundtrips or require additional
+flow control credits issued by the receiver.
```
Comment (marten-seemann) [inline draft-ietf-quic-reliable-stream-reset.md:175]: Should we mention how this works when multiple frames were sent?

File: draft-ietf-quic-reliable-stream-reset.md [hunk 3]
```diff
@@ -158,6 +158,28 @@ a RESET_STREAM_AT frame. Similary, the Reliable Size of the RESET_STREAM_AT
...
+## Stream States {#stream-states}
+
+In terms of stream state transitions ({{Section 3 of RFC9000}}), the effect of
+RESET_STREAM_AT frame is equivalent to that of the FIN bit. This is because both
+the RESET_STREAM_AT frame and the FIN bit serve the same role: signaling the
+amount of data to be delivered.
+
+On the sending side, when a RESET_STREAM_AT frame is sent, the sending part of
+the stream enters the "Data Sent" state. Once the RESET_STREAM_AT frame and all
+stream data up to the Reliable Size are acknowledged, the sending part of the
```
Comment (kazuho) [inline draft-ietf-quic-reliable-stream-reset.md:170]:
```suggestion
On the sending side, when the first RESET_STREAM_AT frame is sent, the sending
part of the stream enters the "Data Sent" state. Once the RESET_STREAM_AT frame
and all stream data up to the smallest Reliable Size being sent are
acknowledged, the sending part of the
```

File: draft-ietf-quic-reliable-stream-reset.md [hunk 4]
```diff
@@ -158,6 +158,28 @@ a RESET_STREAM_AT frame. Similary, the Reliable Size of the RESET_STREAM_AT
...
+stream enters the "Data Recvd" state. Transition from "Data Sent" to "Data
+Recvd" happens immediately when the application resets a stream and if all bytes
+up to the specified Reliable Size have been sent and acknowledged already.
+Conversely, the transition might take multiple roundtrips or require additional
+flow control credits issued by the receiver.
+
+On the receiving side, when a RESET_STREAM_AT frame is received, the receiving
+part of the stream enters the "Size Known" state. Once all data up to the
+Reliable Size have been received, it enters the "Data Recvd" state. Similarly to
```
Comment (kazuho) [inline draft-ietf-quic-reliable-stream-reset.md:179]:
```suggestion
smallest Reliable Size have been received, it enters the "Data Recvd" state.
Similarly to
```

**Review by kazuho — COMMENTED** [review]
@marten-seemann 
> Should we mention how this works when multiple frames were sent?

If you prefer that, following changes might be the way to do it lightly. I do not have a preference, please take it or leave it.

**Review by marten-seemann — APPROVED** [review]

**Pull request merged**